sktime/annotation/hmm.py

```diff
@@ -0,0 +1,314 @@
+# -*- coding: utf-8 -*-
+"""
+HMM Annotation Estimator.
+
+Implements a basic Hidden Markov Model (HMM) as an annotation estimator.
+To read more about the algorithm, check out the `HMM wikipedia page
+<https://en.wikipedia.org/wiki/Hidden_Markov_model>`_.
+"""
+from typing import Tuple
+
+import numpy as np
+
+from sktime.annotation.base._base import BaseSeriesAnnotator
+
+__author__ = ["miraep8"]
+__all__ = ["HMM"]
+
+
+class HMM(BaseSeriesAnnotator):
+    """Implements a simple HMM fitted with Viterbi algorithm.
+
+    The HMM annotation estimator uses the
+    the Viterbi algorithm to fit a sequence of 'hidden state' class
+    annotations (represented by an array of integers the same size
+    as the observation) to a sequence of observations.
+
+    This is done by finding the most likely path given the emission
+    probabilities - (ie the probability that a particular observation
+    would be generated by a given hidden state), the transition prob
+    (ie the probability of transitioning from one state to another or
+    staying in the same state) and the initial probabilities - ie the
+    belief of the probability distribution of hidden states at the
+    start of the observation sequence).
+
+    _fit is currently empty as the parameters of the probability
+    distribution are required to be passed to the algorithm.
+
+    _predict - first the transition_probability and transition_id matrices are
+    calculated - these are both nxm matrices, where n is the number of
+    hidden states and m is the number of observations. The transition
+    probability matrices record the probability of the most likely
+    sequence which has observation `m` being assigned to hidden state n.
+    The transition_id's record the step before hidden state n that
+    proceeds it in the most likely path.  This logic is mostly carried
+    out by helper function _calculate_trans_mats.
+    Next, these matrices are used to calculate the most likely
+    path (by backtracing from the final mostly likely state and the
+    id's that proceeded it.)  This logic is done via a helper func
+    hmm_viterbi_label.
+
+    Parameters
+    ----------
+    emission_funcs : a list of length n.  Should be either a list of
```

Comment:
My suggestions for improvement regarding the docstrings: an example of conventional docs can be found in [boss.py file](https://github.com/alan-turing-institute/sktime/blob/main/sktime/classification/dictionary_based/_boss.py). For example, first line is the type, and broader description of a parameter starts from the next line, which makes it further easier to read. Sth like
```
parameter : float, optional, default=1.5
   Broader description of the parameter.
```

Reply:
Thank you!  It was useful to see a more concrete example of what a good example of docstring looks like :)

Reply:
This looks great, thanks for the changes! 
I think the only docstr left is `_calculate_trans_mats`, and it'll be good to go.

Reply:
Ah yes, good catch!  Will work on that now.  Thanks again for the helpful feedback! :)